service/controller/azuremachinepool/handler/nodepool/delete.go

```diff
@@ -142,13 +142,31 @@ func (r *Resource) deleteARMDeployment(ctx context.Context, azureMachinePool *ca
 
 // deleteVMSS deletes the VMSS from Azure.
 func (r *Resource) deleteVMSS(ctx context.Context, azureMachinePool *capzexpv1alpha3.AzureMachinePool, resourceGroupName, vmssName string) error {
-	r.Logger.LogCtx(ctx, "message", "Deleting machine pool VMSS")
-
 	virtualMachineScaleSetsClient, err := r.ClientFactory.GetVirtualMachineScaleSetsClient(ctx, azureMachinePool.ObjectMeta)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
+	// Delete role assignment related to this VMSS.
```

Comment:
Are we allowed to remove the VMSS first and then the role assignment? It feels safer but not sure if it's possible.

Reply:
It is. Role assignment has a reference to the vmss and not the opposite so it's fine to delete the role assignment first